Error: VALGRIND_WARNING:
bin/ldbdump: note: while executing process 3884 with arguments: -i bin/st/tmp/tmp5sd2zke1/guidindexpackv1.ldb
ldb-2.4.0/bin/default/../../common/ldb.c:105: warning[Leak_PossiblyLost]: 288 bytes in 1 blocks are possibly lost in loss record 22 of 36
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
ldb-2.4.0/bin/default/../../common/ldb.c:105: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbdump.c:327: note: called from main()
#  103|   	}
#  104|   
#  105|-> 	ldb = talloc_zero(mem_ctx, struct ldb_context);
#  106|   	if (ldb == NULL) {
#  107|   		return NULL;

Error: VALGRIND_WARNING:
bin/ldbadd: note: while executing process 2503
ldb-2.4.0/bin/default/../../common/ldb.c:114: warning[Leak_PossiblyLost]: 128 bytes in 1 blocks are possibly lost in loss record 6 of 26
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_named_const()
/usr/lib64/libtevent.so.0.11.0: note: called from tevent_register_backend()
/usr/lib64/libtevent.so.0.11.0: note: called from here
/usr/lib64/libtevent.so.0.11.0: note: called from here
/usr/lib64/libtevent.so.0.11.0: note: called from tevent_context_init_byname()
ldb-2.4.0/bin/default/../../common/ldb.c:114: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbadd.c:138: note: called from main()
#  112|   	 * having to provide their own private one explicitly */
#  113|   	if (ev_ctx == NULL) {
#  114|-> 		ev_ctx = tevent_context_init(ldb);
#  115|   		if (ev_ctx == NULL) {
#  116|   			talloc_free(ldb);

Error: VALGRIND_WARNING:
bin/ldbdump: note: while executing process 3884 with arguments: -i bin/st/tmp/tmp5sd2zke1/guidindexpackv1.ldb
ldb-2.4.0/bin/default/../../common/ldb.c:114: warning[Leak_PossiblyLost]: 136 bytes in 1 blocks are possibly lost in loss record 15 of 36
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
/usr/lib64/libtevent.so.0.11.0: note: called from here
/usr/lib64/libtevent.so.0.11.0: note: called from here
/usr/lib64/libtevent.so.0.11.0: note: called from tevent_context_init_ops()
ldb-2.4.0/bin/default/../../common/ldb.c:114: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbdump.c:327: note: called from main()
#  112|   	 * having to provide their own private one explicitly */
#  113|   	if (ev_ctx == NULL) {
#  114|-> 		ev_ctx = tevent_context_init(ldb);
#  115|   		if (ev_ctx == NULL) {
#  116|   			talloc_free(ldb);

Error: VALGRIND_WARNING:
bin/ldbdump: note: while executing process 3884 with arguments: -i bin/st/tmp/tmp5sd2zke1/guidindexpackv1.ldb
ldb-2.4.0/bin/default/../../common/ldb.c:114: warning[Leak_PossiblyLost]: 176 bytes in 1 blocks are possibly lost in loss record 18 of 36
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
/usr/lib64/libtevent.so.0.11.0: note: called from here
/usr/lib64/libtevent.so.0.11.0: note: called from tevent_context_init_ops()
ldb-2.4.0/bin/default/../../common/ldb.c:114: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbdump.c:327: note: called from main()
#  112|   	 * having to provide their own private one explicitly */
#  113|   	if (ev_ctx == NULL) {
#  114|-> 		ev_ctx = tevent_context_init(ldb);
#  115|   		if (ev_ctx == NULL) {
#  116|   			talloc_free(ldb);

Error: VALGRIND_WARNING:
bin/ldbdump: note: while executing process 3884 with arguments: -i bin/st/tmp/tmp5sd2zke1/guidindexpackv1.ldb
ldb-2.4.0/bin/default/../../common/ldb.c:114: warning[Leak_PossiblyLost]: 376 bytes in 1 blocks are possibly lost in loss record 29 of 36
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
/usr/lib64/libtevent.so.0.11.0: note: called from tevent_context_init_ops()
ldb-2.4.0/bin/default/../../common/ldb.c:114: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbdump.c:327: note: called from main()
#  112|   	 * having to provide their own private one explicitly */
#  113|   	if (ev_ctx == NULL) {
#  114|-> 		ev_ctx = tevent_context_init(ldb);
#  115|   		if (ev_ctx == NULL) {
#  116|   			talloc_free(ldb);

Error: VALGRIND_WARNING:
bin/ldbdump: note: while executing process 3884 with arguments: -i bin/st/tmp/tmp5sd2zke1/guidindexpackv1.ldb
ldb-2.4.0/bin/default/../../common/ldb_attributes.c:96: warning[Leak_PossiblyLost]: 240 bytes in 1 blocks are possibly lost in loss record 20 of 36
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_realloc()
ldb-2.4.0/bin/default/../../common/ldb_attributes.c:96: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_attributes.c:82: note: called from ldb_schema_attribute_add_with_syntax()
ldb-2.4.0/bin/default/../../common/ldb_attributes.c:318: note: called from ldb_setup_wellknown_attributes()
ldb-2.4.0/bin/default/../../common/ldb.c:123: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbdump.c:327: note: called from main()
#   94|   	n = ldb->schema.num_attributes + 1;
#   95|   
#   96|-> 	a = talloc_realloc(ldb, ldb->schema.attributes,
#   97|   			   struct ldb_schema_attribute, n);
#   98|   	if (a == NULL) {

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_attributes.c:345: warning[Leak_PossiblyLost]: 128 bytes in 1 blocks are possibly lost in loss record 62 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_realloc()
ldb-2.4.0/bin/default/../../common/ldb_attributes.c:345: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_attributes.c:332: note: called from ldb_dn_extended_add_syntax()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:181: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  343|   	n = ldb->schema.num_dn_extended_syntax + 1;
#  344|   
#  345|-> 	a = talloc_realloc(ldb, ldb->schema.dn_extended_syntax,
#  346|   			   struct ldb_dn_extended_syntax, n);
#  347|   

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:104: warning[Leak_PossiblyLost]: 168 bytes in 1 blocks are possibly lost in loss record 75 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:104: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:89: note: called from ldb_dn_from_ldb_val()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:165: note: called from ldb_dn_new()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:33: note: called from test_ldb_dn_add_child_fmt()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  102|   	}
#  103|   
#  104|-> 	dn = talloc_zero(mem_ctx, struct ldb_dn);
#  105|   	LDB_DN_NULL_FAILED(dn);
#  106|   

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:104: warning[Leak_PossiblyLost]: 2,352 bytes in 14 blocks are possibly lost in loss record 140 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:104: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:89: note: called from ldb_dn_from_ldb_val()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:165: note: called from ldb_dn_new()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:187: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  102|   	}
#  103|   
#  104|-> 	dn = talloc_zero(mem_ctx, struct ldb_dn);
#  105|   	LDB_DN_NULL_FAILED(dn);
#  106|   

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:122: warning[Leak_PossiblyLost]: 1,350 bytes in 13 blocks are possibly lost in loss record 138 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strndup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:122: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:89: note: called from ldb_dn_from_ldb_val()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:165: note: called from ldb_dn_new()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:187: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  120|   			dn->special = true;
#  121|   		}
#  122|-> 		dn->ext_linearized = talloc_strndup(dn, data, length);
#  123|   		LDB_DN_NULL_FAILED(dn->ext_linearized);
#  124|   

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:122: warning[Leak_PossiblyLost]: 100 bytes in 1 blocks are possibly lost in loss record 9 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strndup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:122: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:89: note: called from ldb_dn_from_ldb_val()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:165: note: called from ldb_dn_new()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1642: note: called from ldb_dn_add_child_val()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:75: note: called from test_ldb_dn_add_child_val()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  120|   			dn->special = true;
#  121|   		}
#  122|-> 		dn->ext_linearized = talloc_strndup(dn, data, length);
#  123|   		LDB_DN_NULL_FAILED(dn->ext_linearized);
#  124|   

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:136: warning[Leak_PossiblyLost]: 485 bytes in 5 blocks are possibly lost in loss record 131 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:136: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:89: note: called from ldb_dn_from_ldb_val()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:165: note: called from ldb_dn_new()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:187: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  134|   
#  135|   			if (p_save == dn->ext_linearized) {
#  136|-> 				dn->linearized = talloc_strdup(dn, "");
#  137|   			} else {
#  138|   				dn->linearized = talloc_strdup(dn, p_save);

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:138: warning[Leak_PossiblyLost]: 409 bytes in 4 blocks are possibly lost in loss record 130 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:138: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:89: note: called from ldb_dn_from_ldb_val()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:165: note: called from ldb_dn_new()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:187: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  136|   				dn->linearized = talloc_strdup(dn, "");
#  137|   			} else {
#  138|-> 				dn->linearized = talloc_strdup(dn, p_save);
#  139|   			}
#  140|   			LDB_DN_NULL_FAILED(dn->linearized);

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:337: warning[Leak_PossiblyLost]: 1,200 bytes in 5 blocks are possibly lost in loss record 136 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:337: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:837: note: called from ldb_dn_get_extended_linearized()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:198: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  335|   	/* in the common case we have 3 or more components */
#  336|   	/* make sure all components are zeroed, other functions depend on it */
#  337|-> 	dn->components = talloc_zero_array(dn, struct ldb_dn_component, 3);
#  338|   	if (dn->components == NULL) {
#  339|   		return false;

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:337: warning[Leak_PossiblyLost]: 240 bytes in 1 blocks are possibly lost in loss record 92 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:337: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:305: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:941: note: called from ldb_dn_casefold_internal()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:993: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:41: note: called from test_ldb_dn_add_child_fmt()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  335|   	/* in the common case we have 3 or more components */
#  336|   	/* make sure all components are zeroed, other functions depend on it */
#  337|-> 	dn->components = talloc_zero_array(dn, struct ldb_dn_component, 3);
#  338|   	if (dn->components == NULL) {
#  339|   		return false;

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:337: warning[Leak_PossiblyLost]: 720 bytes in 3 blocks are possibly lost in loss record 135 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:337: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:210: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  335|   	/* in the common case we have 3 or more components */
#  336|   	/* make sure all components are zeroed, other functions depend on it */
#  337|-> 	dn->components = talloc_zero_array(dn, struct ldb_dn_component, 3);
#  338|   	if (dn->components == NULL) {
#  339|   		return false;

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:394: warning[Leak_PossiblyLost]: 120 bytes in 1 blocks are possibly lost in loss record 49 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_realloc()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:394: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:837: note: called from ldb_dn_get_extended_linearized()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:198: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  392|   				/* Process name and ex_value */
#  393|   
#  394|-> 				ext_comp = talloc_realloc(
#  395|   					dn,
#  396|   					dn->ext_components,

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:490: warning[Leak_PossiblyLost]: 196 bytes in 2 blocks are possibly lost in loss record 87 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:490: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:837: note: called from ldb_dn_get_extended_linearized()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:198: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  488|   				 *  with spaces trimmed) */
#  489|   				*d++ = '\0';
#  490|-> 				dn->components[dn->comp_num].name = talloc_strdup(dn->components, dt);
#  491|   				if (dn->components[dn->comp_num].name == NULL) {
#  492|   					/* ouch */

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:490: warning[Leak_PossiblyLost]: 198 bytes in 2 blocks are possibly lost in loss record 88 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:490: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1956: note: called from ldb_dn_set_component()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1649: note: called from ldb_dn_add_child_val()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:75: note: called from test_ldb_dn_add_child_val()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  488|   				 *  with spaces trimmed) */
#  489|   				*d++ = '\0';
#  490|-> 				dn->components[dn->comp_num].name = talloc_strdup(dn->components, dt);
#  491|   				if (dn->components[dn->comp_num].name == NULL) {
#  492|   					/* ouch */

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:490: warning[Leak_PossiblyLost]: 297 bytes in 3 blocks are possibly lost in loss record 106 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:490: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:305: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:941: note: called from ldb_dn_casefold_internal()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:993: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:41: note: called from test_ldb_dn_add_child_fmt()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  488|   				 *  with spaces trimmed) */
#  489|   				*d++ = '\0';
#  490|-> 				dn->components[dn->comp_num].name = talloc_strdup(dn->components, dt);
#  491|   				if (dn->components[dn->comp_num].name == NULL) {
#  492|   					/* ouch */

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:490: warning[Leak_PossiblyLost]: 392 bytes in 4 blocks are possibly lost in loss record 126 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:490: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:210: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  488|   				 *  with spaces trimmed) */
#  489|   				*d++ = '\0';
#  490|-> 				dn->components[dn->comp_num].name = talloc_strdup(dn->components, dt);
#  491|   				if (dn->components[dn->comp_num].name == NULL) {
#  492|   					/* ouch */

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:490: warning[Leak_PossiblyLost]: 396 bytes in 4 blocks are possibly lost in loss record 127 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:490: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:305: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:941: note: called from ldb_dn_casefold_internal()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:993: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:58: note: called from test_ldb_dn_add_child_fmt2()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  488|   				 *  with spaces trimmed) */
#  489|   				*d++ = '\0';
#  490|-> 				dn->components[dn->comp_num].name = talloc_strdup(dn->components, dt);
#  491|   				if (dn->components[dn->comp_num].name == NULL) {
#  492|   					/* ouch */

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:589: warning[Leak_PossiblyLost]: 102 bytes in 1 blocks are possibly lost in loss record 14 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_memdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:589: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1956: note: called from ldb_dn_set_component()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1649: note: called from ldb_dn_add_child_val()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:75: note: called from test_ldb_dn_add_child_val()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  587|   				 */
#  588|   				dn->components[dn->comp_num].value.data = \
#  589|-> 					(uint8_t *)talloc_memdup(dn->components, dt, l + 1);
#  590|   				dn->components[dn->comp_num].value.length = l;
#  591|   				if (dn->components[dn->comp_num].value.data == NULL) {

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:589: warning[Leak_PossiblyLost]: 200 bytes in 2 blocks are possibly lost in loss record 90 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_memdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:589: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:305: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:941: note: called from ldb_dn_casefold_internal()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:993: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:41: note: called from test_ldb_dn_add_child_fmt()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  587|   				 */
#  588|   				dn->components[dn->comp_num].value.data = \
#  589|-> 					(uint8_t *)talloc_memdup(dn->components, dt, l + 1);
#  590|   				dn->components[dn->comp_num].value.length = l;
#  591|   				if (dn->components[dn->comp_num].value.data == NULL) {

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:589: warning[Leak_PossiblyLost]: 298 bytes in 3 blocks are possibly lost in loss record 111 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_memdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:589: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:305: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:941: note: called from ldb_dn_casefold_internal()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:993: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:58: note: called from test_ldb_dn_add_child_fmt2()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  587|   				 */
#  588|   				dn->components[dn->comp_num].value.data = \
#  589|-> 					(uint8_t *)talloc_memdup(dn->components, dt, l + 1);
#  590|   				dn->components[dn->comp_num].value.length = l;
#  591|   				if (dn->components[dn->comp_num].value.data == NULL) {

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:589: warning[Leak_PossiblyLost]: 98 bytes in 1 blocks are possibly lost in loss record 2 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_memdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:589: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:210: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  587|   				 */
#  588|   				dn->components[dn->comp_num].value.data = \
#  589|-> 					(uint8_t *)talloc_memdup(dn->components, dt, l + 1);
#  590|   				dn->components[dn->comp_num].value.length = l;
#  591|   				if (dn->components[dn->comp_num].value.data == NULL) {

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:602: warning[Leak_PossiblyLost]: 288 bytes in 1 blocks are possibly lost in loss record 102 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_realloc()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:602: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:305: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:941: note: called from ldb_dn_casefold_internal()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:993: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:58: note: called from test_ldb_dn_add_child_fmt2()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  600|   				dn->comp_num++;
#  601|   				if (dn->comp_num > 2) {
#  602|-> 					dn->components = talloc_realloc(dn,
#  603|   									dn->components,
#  604|   									struct ldb_dn_component,

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:731: warning[Leak_PossiblyLost]: 100 bytes in 1 blocks are possibly lost in loss record 7 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_memdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:731: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:305: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:941: note: called from ldb_dn_casefold_internal()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:993: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:41: note: called from test_ldb_dn_add_child_fmt()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  729|   		dn->components[dn->comp_num].value.length = l;
#  730|   		dn->components[dn->comp_num].value.data =
#  731|-> 			(uint8_t *)talloc_memdup(dn->components, dt, l + 1);
#  732|   		if (dn->components[dn->comp_num].value.data == NULL) {
#  733|   			/* ouch */

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:731: warning[Leak_PossiblyLost]: 297 bytes in 3 blocks are possibly lost in loss record 110 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_memdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:731: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:210: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  729|   		dn->components[dn->comp_num].value.length = l;
#  730|   		dn->components[dn->comp_num].value.data =
#  731|-> 			(uint8_t *)talloc_memdup(dn->components, dt, l + 1);
#  732|   		if (dn->components[dn->comp_num].value.data == NULL) {
#  733|   			/* ouch */

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:731: warning[Leak_PossiblyLost]: 98 bytes in 1 blocks are possibly lost in loss record 4 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_memdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:731: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:837: note: called from ldb_dn_get_extended_linearized()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:198: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  729|   		dn->components[dn->comp_num].value.length = l;
#  730|   		dn->components[dn->comp_num].value.data =
#  731|-> 			(uint8_t *)talloc_memdup(dn->components, dt, l + 1);
#  732|   		if (dn->components[dn->comp_num].value.data == NULL) {
#  733|   			/* ouch */

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:787: warning[Leak_PossiblyLost]: 135 bytes in 1 blocks are possibly lost in loss record 64 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_array()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:787: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:758: note: called from ldb_dn_get_linearized()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:78: note: called from test_ldb_dn_add_child_val()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  785|   		len += 2; /* '=' and ',' */
#  786|   	}
#  787|-> 	dn->linearized = talloc_array(dn, char, len);
#  788|   	if ( ! dn->linearized) return NULL;
#  789|   

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:787: warning[Leak_PossiblyLost]: 150 bytes in 1 blocks are possibly lost in loss record 72 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_array()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:787: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:758: note: called from ldb_dn_get_linearized()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:99: note: called from test_ldb_dn_add_child_val2()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  785|   		len += 2; /* '=' and ',' */
#  786|   	}
#  787|-> 	dn->linearized = talloc_array(dn, char, len);
#  788|   	if ( ! dn->linearized) return NULL;
#  789|   

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1010: warning[Leak_PossiblyLost]: 135 bytes in 1 blocks are possibly lost in loss record 63 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_array()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1010: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:978: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:41: note: called from test_ldb_dn_add_child_fmt()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
# 1008|   		len += 2; /* '=' and ',' */
# 1009|   	}
# 1010|-> 	dn->casefold = talloc_array(dn, char, len);
# 1011|   	if ( ! dn->casefold) return NULL;
# 1012|   

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1010: warning[Leak_PossiblyLost]: 142 bytes in 1 blocks are possibly lost in loss record 71 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_array()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1010: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:978: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:58: note: called from test_ldb_dn_add_child_fmt2()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
# 1008|   		len += 2; /* '=' and ',' */
# 1009|   	}
# 1010|-> 	dn->casefold = talloc_array(dn, char, len);
# 1011|   	if ( ! dn->casefold) return NULL;
# 1012|   

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1010: warning[Leak_PossiblyLost]: 150 bytes in 1 blocks are possibly lost in loss record 73 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_array()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1010: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:978: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:102: note: called from test_ldb_dn_add_child_val2()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
# 1008|   		len += 2; /* '=' and ',' */
# 1009|   	}
# 1010|-> 	dn->casefold = talloc_array(dn, char, len);
# 1011|   	if ( ! dn->casefold) return NULL;
# 1012|   

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1574: warning[Leak_PossiblyLost]: 117 bytes in 1 blocks are possibly lost in loss record 33 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_vasprintf()
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_asprintf()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1574: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1501: note: called from ldb_dn_add_child()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1618: note: called from ldb_dn_add_child_fmt()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:35: note: called from test_ldb_dn_add_child_fmt()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
# 1572|   		}
# 1573|   
# 1574|-> 		t = talloc_asprintf(dn, "%s,%s", s, dn->linearized);
# 1575|   		if ( ! t) {
# 1576|   			ldb_dn_mark_invalid(dn);

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1574: warning[Leak_PossiblyLost]: 122 bytes in 1 blocks are possibly lost in loss record 50 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_vasprintf()
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_asprintf()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1574: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1501: note: called from ldb_dn_add_child()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1618: note: called from ldb_dn_add_child_fmt()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:52: note: called from test_ldb_dn_add_child_fmt2()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
# 1572|   		}
# 1573|   
# 1574|-> 		t = talloc_asprintf(dn, "%s,%s", s, dn->linearized);
# 1575|   		if ( ! t) {
# 1576|   			ldb_dn_mark_invalid(dn);

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1972: warning[Leak_PossiblyLost]: 99 bytes in 1 blocks are possibly lost in loss record 5 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strdup()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1972: note: called from ldb_dn_set_component()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1649: note: called from ldb_dn_add_child_val()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:75: note: called from test_ldb_dn_add_child_val()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
# 1970|   	}
# 1971|   
# 1972|-> 	n = talloc_strdup(dn, name);
# 1973|   	if ( ! n) {
# 1974|   		return LDB_ERR_OTHER;

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1983: warning[Leak_PossiblyLost]: 103 bytes in 1 blocks are possibly lost in loss record 16 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_named_const()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1983: note: called from ldb_dn_set_component()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1649: note: called from ldb_dn_add_child_val()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:96: note: called from test_ldb_dn_add_child_val2()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
# 1981|   	 * avoids the over-read
# 1982|   	 */
# 1983|-> 	v.data = (uint8_t *)talloc_size(dn, v.length+1);
# 1984|   	if ( ! v.data) {
# 1985|   		talloc_free(n);

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1983: warning[Leak_PossiblyLost]: 98 bytes in 1 blocks are possibly lost in loss record 1 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_named_const()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1983: note: called from ldb_dn_set_component()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:1649: note: called from ldb_dn_add_child_val()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:75: note: called from test_ldb_dn_add_child_val()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
# 1981|   	 * avoids the over-read
# 1982|   	 */
# 1983|-> 	v.data = (uint8_t *)talloc_size(dn, v.length+1);
# 1984|   	if ( ! v.data) {
# 1985|   		talloc_free(n);

Error: VALGRIND_WARNING:
bin/ldbdump: note: while executing process 3884 with arguments: -i bin/st/tmp/tmp5sd2zke1/guidindexpackv1.ldb
ldb-2.4.0/bin/default/../../common/ldb_match.c:676: warning[Leak_PossiblyLost]: 112 bytes in 1 blocks are possibly lost in loss record 2 of 36
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
ldb-2.4.0/bin/default/../../common/ldb_match.c:676: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb.c:133: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbdump.c:327: note: called from main()
#  674|   
#  675|   	/* Register bitmask-and match */
#  676|-> 	bitmask_and = talloc_zero(ldb, struct ldb_extended_match_rule);
#  677|   	if (bitmask_and == NULL) {
#  678|   		return LDB_ERR_OPERATIONS_ERROR;

Error: VALGRIND_WARNING:
bin/ldbdump: note: while executing process 3884 with arguments: -i bin/st/tmp/tmp5sd2zke1/guidindexpackv1.ldb
ldb-2.4.0/bin/default/../../common/ldb_match.c:690: warning[Leak_PossiblyLost]: 112 bytes in 1 blocks are possibly lost in loss record 3 of 36
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
ldb-2.4.0/bin/default/../../common/ldb_match.c:690: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_match.c:668: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb.c:133: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbdump.c:327: note: called from main()
#  688|   
#  689|   	/* Register bitmask-or match */
#  690|-> 	bitmask_or = talloc_zero(ldb, struct ldb_extended_match_rule);
#  691|   	if (bitmask_or == NULL) {
#  692|   		return LDB_ERR_OPERATIONS_ERROR;

Error: VALGRIND_WARNING:
bin/ldbdump: note: while executing process 3884 with arguments: -i bin/st/tmp/tmp5sd2zke1/guidindexpackv1.ldb
ldb-2.4.0/bin/default/../../common/ldb_match.c:704: warning[Leak_PossiblyLost]: 112 bytes in 1 blocks are possibly lost in loss record 4 of 36
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
ldb-2.4.0/bin/default/../../common/ldb_match.c:704: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_match.c:668: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb.c:133: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbdump.c:327: note: called from main()
#  702|   
#  703|   	/* Register always-false match */
#  704|-> 	always_false = talloc_zero(ldb, struct ldb_extended_match_rule);
#  705|   	if (always_false == NULL) {
#  706|   		return LDB_ERR_OPERATIONS_ERROR;

Error: VALGRIND_WARNING:
bin/ldbdump: note: while executing process 3884 with arguments: -i bin/st/tmp/tmp5sd2zke1/guidindexpackv1.ldb
ldb-2.4.0/bin/default/../../common/ldb_match.c:734: warning[Leak_PossiblyLost]: 120 bytes in 1 blocks are possibly lost in loss record 5 of 36
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
ldb-2.4.0/bin/default/../../common/ldb_match.c:734: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_match.c:723: note: called from ldb_register_extended_match_rule()
ldb-2.4.0/bin/default/../../common/ldb_match.c:684: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb.c:133: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbdump.c:327: note: called from main()
#  732|   	}
#  733|   
#  734|-> 	entry = talloc_zero(ldb, struct ldb_extended_match_entry);
#  735|   	if (!entry) {
#  736|   		return LDB_ERR_OPERATIONS_ERROR;

Error: VALGRIND_WARNING:
bin/ldbadd: note: while executing process 2503
ldb-2.4.0/bin/default/../../common/ldb_modules.c:156: warning[Leak_PossiblyLost]: 360 bytes in 3 blocks are possibly lost in loss record 15 of 26
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:156: note: called from ldb_register_backend()
ldb-2.4.0/bin/default/../../ldb_ldap/ldb_ldap.c:961: note: called from ldb_init_module()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:1017: note: called from ldb_modules_load_path()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:1091: note: called from ldb_modules_load_dir()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:976: note: called from ldb_modules_load_path()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:1167: note: called from ldb_modules_load()
ldb-2.4.0/bin/default/../../common/ldb.c:100: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbadd.c:138: note: called from main()
#  154|   		}
#  155|   	} else {
#  156|-> 		be = talloc_zero(ldb_backends, struct backends_list_entry);
#  157|   		if (!be) {
#  158|   			return LDB_ERR_OPERATIONS_ERROR;

Error: VALGRIND_WARNING:
bin/ldbadd: note: while executing process 2503
ldb-2.4.0/bin/default/../../common/ldb_modules.c:160: warning[Leak_PossiblyLost]: 336 bytes in 3 blocks are possibly lost in loss record 13 of 26
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_zero()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:160: note: called from ldb_register_backend()
ldb-2.4.0/bin/default/../../ldb_ldap/ldb_ldap.c:961: note: called from ldb_init_module()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:1017: note: called from ldb_modules_load_path()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:1091: note: called from ldb_modules_load_dir()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:976: note: called from ldb_modules_load_path()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:1167: note: called from ldb_modules_load()
ldb-2.4.0/bin/default/../../common/ldb.c:100: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbadd.c:138: note: called from main()
#  158|   			return LDB_ERR_OPERATIONS_ERROR;
#  159|   		}
#  160|-> 		be->ops = talloc_zero(be, struct ldb_backend_ops);
#  161|   		if (!be->ops) {
#  162|   			talloc_free(be);

Error: VALGRIND_WARNING:
bin/ldbadd: note: while executing process 2503
ldb-2.4.0/bin/default/../../common/ldb_modules.c:297: warning[Leak_PossiblyLost]: 672 bytes in 6 blocks are possibly lost in loss record 20 of 26
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_named_const()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:297: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:280: note: called from ldb_register_module()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:1017: note: called from ldb_modules_load_path()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:1091: note: called from ldb_modules_load_dir()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:976: note: called from ldb_modules_load_path()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:1167: note: called from ldb_modules_load()
ldb-2.4.0/bin/default/../../common/ldb.c:100: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbadd.c:138: note: called from main()
#  295|   	 * frees the list.
#  296|   	 */
#  297|-> 	entry = talloc(NULL, struct ops_list_entry);
#  298|   	if (entry == NULL) {
#  299|   		return LDB_ERR_OPERATIONS_ERROR;

Error: VALGRIND_WARNING:
bin/ldbadd: note: while executing process 2503
ldb-2.4.0/bin/default/../../common/ldb_modules.c:963: warning[Leak_PossiblyLost]: 1,280 bytes in 10 blocks are possibly lost in loss record 22 of 26
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_named_const()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:963: note: called from ldb_modules_load_path()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:1091: note: called from ldb_modules_load_dir()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:976: note: called from ldb_modules_load_path()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:1167: note: called from ldb_modules_load()
ldb-2.4.0/bin/default/../../common/ldb.c:100: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbadd.c:138: note: called from main()
#  961|   	}
#  962|   
#  963|-> 	le = talloc(loaded, struct loaded);
#  964|   	if (le == NULL) {
#  965|   		fprintf(stderr, "ldb: unable to allocated loaded entry\n");

Error: VALGRIND_WARNING:
bin/ldbadd: note: while executing process 2503
ldb-2.4.0/bin/default/../../common/ldb_modules.c:963: warning[Leak_PossiblyLost]: 128 bytes in 1 blocks are possibly lost in loss record 5 of 26
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_named_const()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:963: note: called from ldb_modules_load_path()
ldb-2.4.0/bin/default/../../common/ldb_modules.c:1167: note: called from ldb_modules_load()
ldb-2.4.0/bin/default/../../common/ldb.c:100: note: called from ldb_init()
ldb-2.4.0/bin/default/../../tools/ldbadd.c:138: note: called from main()
#  961|   	}
#  962|   
#  963|-> 	le = talloc(loaded, struct loaded);
#  964|   	if (le == NULL) {
#  965|   		fprintf(stderr, "ldb: unable to allocated loaded entry\n");

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_utf8.c:59: warning[Leak_PossiblyLost]: 300 bytes in 3 blocks are possibly lost in loss record 112 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strndup()
ldb-2.4.0/bin/default/../../common/ldb_utf8.c:59: note: called from ldb_casefold_default()
ldb-2.4.0/bin/default/../../common/attrib_handlers.c:64: note: called from ldb_handler_fold()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:958: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:932: note: called from ldb_dn_casefold_internal()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:993: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:41: note: called from test_ldb_dn_add_child_fmt()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#   57|   {
#   58|   	size_t i;
#   59|-> 	char *ret = talloc_strndup(mem_ctx, s, n);
#   60|   	if (!s) {
#   61|   		errno = ENOMEM;

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_utf8.c:59: warning[Leak_PossiblyLost]: 305 bytes in 3 blocks are possibly lost in loss record 114 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strndup()
ldb-2.4.0/bin/default/../../common/ldb_utf8.c:59: note: called from ldb_casefold_default()
ldb-2.4.0/bin/default/../../common/attrib_handlers.c:64: note: called from ldb_handler_fold()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:958: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:932: note: called from ldb_dn_casefold_internal()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:993: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:102: note: called from test_ldb_dn_add_child_val2()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#   57|   {
#   58|   	size_t i;
#   59|-> 	char *ret = talloc_strndup(mem_ctx, s, n);
#   60|   	if (!s) {
#   61|   		errno = ENOMEM;

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_utf8.c:59: warning[Leak_PossiblyLost]: 398 bytes in 4 blocks are possibly lost in loss record 129 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strndup()
ldb-2.4.0/bin/default/../../common/ldb_utf8.c:59: note: called from ldb_casefold_default()
ldb-2.4.0/bin/default/../../common/attrib_handlers.c:64: note: called from ldb_handler_fold()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:958: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:932: note: called from ldb_dn_casefold_internal()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:993: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:58: note: called from test_ldb_dn_add_child_fmt2()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#   57|   {
#   58|   	size_t i;
#   59|-> 	char *ret = talloc_strndup(mem_ctx, s, n);
#   60|   	if (!s) {
#   61|   		errno = ENOMEM;

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_utf8.c:115: warning[Leak_PossiblyLost]: 297 bytes in 3 blocks are possibly lost in loss record 107 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strdup()
ldb-2.4.0/bin/default/../../common/ldb_utf8.c:115: note: called from ldb_attr_casefold()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:949: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:932: note: called from ldb_dn_casefold_internal()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:993: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:41: note: called from test_ldb_dn_add_child_fmt()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  113|   {
#  114|   	size_t i;
#  115|-> 	char *ret = talloc_strdup(mem_ctx, s);
#  116|   	if (!ret) {
#  117|   		errno = ENOMEM;

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../common/ldb_utf8.c:115: warning[Leak_PossiblyLost]: 396 bytes in 4 blocks are possibly lost in loss record 128 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strdup()
ldb-2.4.0/bin/default/../../common/ldb_utf8.c:115: note: called from ldb_attr_casefold()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:949: note: called from UnknownInlinedFun()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:932: note: called from ldb_dn_casefold_internal()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:993: note: called from ldb_dn_get_casefold()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:58: note: called from test_ldb_dn_add_child_fmt2()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  113|   {
#  114|   	size_t i;
#  115|-> 	char *ret = talloc_strdup(mem_ctx, s);
#  116|   	if (!ret) {
#  117|   		errno = ENOMEM;

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:131: warning[Leak_PossiblyLost]: 100 bytes in 1 blocks are possibly lost in loss record 13 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_memdup()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:131: note: called from extended_dn_read_ID()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:414: note: called from ldb_dn_explode.part.0()
ldb-2.4.0/bin/default/../../common/ldb_dn.c:837: note: called from ldb_dn_get_extended_linearized()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:198: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  129|   
#  130|   	*out = *in;
#  131|-> 	out->data = talloc_memdup(mem_ctx, in->data, in->length);
#  132|   	if (out->data == NULL) {
#  133|   		return -1;

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:198: warning[Leak_PossiblyLost]: 114 bytes in 1 blocks are possibly lost in loss record 32 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from _talloc_realloc()
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_asprintf_append_buffer()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:198: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  196|   				    tests[i].linearized);
#  197|   
#  198|-> 		ext_linear = ldb_dn_get_extended_linearized(ldb, dn, 1);
#  199|   		assert_true((ext_linear == NULL) ==
#  200|   			    (tests[i].ext_linearized_1 == NULL));

Error: VALGRIND_WARNING:
bin/test_ldb_dn: note: while executing process 7246
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:198: warning[Leak_PossiblyLost]: 603 bytes in 6 blocks are possibly lost in loss record 132 of 143
/usr/lib64/libtalloc.so.2.3.3: note: called from here
/usr/lib64/libtalloc.so.2.3.3: note: called from talloc_strdup()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:198: note: called from test_ldb_dn_explode()
/usr/lib64/libcmocka.so.0.7.0: note: called from here
/usr/lib64/libcmocka.so.0.7.0: note: called from _cmocka_run_group_tests()
ldb-2.4.0/bin/default/../../tests/test_ldb_dn.c:231: note: called from main()
#  196|   				    tests[i].linearized);
#  197|   
#  198|-> 		ext_linear = ldb_dn_get_extended_linearized(ldb, dn, 1);
#  199|   		assert_true((ext_linear == NULL) ==
#  200|   			    (tests[i].ext_linearized_1 == NULL));

Error: VALGRIND_WARNING:
bin/ldbadd: note: while executing process 2503
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: warning[Leak_DefinitelyLost]: 815 (792 direct, 23 indirect) bytes in 1 blocks are definitely lost in loss record 21 of 26
/usr/lib64/libpopt.so.0.0.1: note: called from poptGetContext()
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: note: called from ldb_cmdline_process_internal.constprop.0()
ldb-2.4.0/bin/default/../../tools/ldbadd.c:143: note: called from main()
#  342|   	}
#  343|   
#  344|-> 	pc = poptGetContext(argv[0], argc, argv, *popt_options,
#  345|   			    POPT_CONTEXT_KEEP_FIRST);
#  346|   

Error: VALGRIND_WARNING:
bin/ldbadd: note: while executing process 2601 with arguments: tests/../tests/init.ldif
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: warning[Leak_DefinitelyLost]: 831 (792 direct, 39 indirect) bytes in 1 blocks are definitely lost in loss record 21 of 26
/usr/lib64/libpopt.so.0.0.1: note: called from poptGetContext()
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: note: called from ldb_cmdline_process_internal.constprop.0()
ldb-2.4.0/bin/default/../../tools/ldbadd.c:143: note: called from main()
#  342|   	}
#  343|   
#  344|-> 	pc = poptGetContext(argv[0], argc, argv, *popt_options,
#  345|   			    POPT_CONTEXT_KEEP_FIRST);
#  346|   

Error: VALGRIND_WARNING:
bin/ldbmodify: note: while executing process 3091 with arguments: tests/../tests/test-modify.ldif
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: warning[Leak_DefinitelyLost]: 834 (792 direct, 42 indirect) bytes in 1 blocks are definitely lost in loss record 21 of 26
/usr/lib64/libpopt.so.0.0.1: note: called from poptGetContext()
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: note: called from ldb_cmdline_process_internal.constprop.0()
ldb-2.4.0/bin/default/../../tools/ldbmodify.c:160: note: called from main()
#  342|   	}
#  343|   
#  344|-> 	pc = poptGetContext(argv[0], argc, argv, *popt_options,
#  345|   			    POPT_CONTEXT_KEEP_FIRST);
#  346|   

Error: VALGRIND_WARNING:
bin/ldbrename: note: while executing process 3581 with arguments: cn=Ursula Hampster,ou=Alumni Association,ou=People,o=University of Michigan,c=TEST cn=Hampster Ursula,ou=Alumni Association,ou=People,o=University of Michigan,c=TEST
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: warning[Leak_DefinitelyLost]: 850 (792 direct, 58 indirect) bytes in 1 blocks are definitely lost in loss record 21 of 26
/usr/lib64/libpopt.so.0.0.1: note: called from poptGetContext()
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: note: called from ldb_cmdline_process_internal.constprop.0()
ldb-2.4.0/bin/default/../../tools/ldbrename.c:62: note: called from main()
#  342|   	}
#  343|   
#  344|-> 	pc = poptGetContext(argv[0], argc, argv, *popt_options,
#  345|   			    POPT_CONTEXT_KEEP_FIRST);
#  346|   

Error: VALGRIND_WARNING:
bin/ldbadd: note: while executing process 2503
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: warning[Leak_PossiblyLost]: 16 bytes in 1 blocks are possibly lost in loss record 2 of 26
/usr/lib64/libpopt.so.0.0.1: note: called from poptGetContext()
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: note: called from ldb_cmdline_process_internal.constprop.0()
ldb-2.4.0/bin/default/../../tools/ldbadd.c:143: note: called from main()
#  342|   	}
#  343|   
#  344|-> 	pc = poptGetContext(argv[0], argc, argv, *popt_options,
#  345|   			    POPT_CONTEXT_KEEP_FIRST);
#  346|   

Error: VALGRIND_WARNING:
bin/ldbadd: note: while executing process 2601 with arguments: tests/../tests/init.ldif
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: warning[Leak_PossiblyLost]: 24 bytes in 1 blocks are possibly lost in loss record 2 of 26
/usr/lib64/libpopt.so.0.0.1: note: called from poptGetContext()
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: note: called from ldb_cmdline_process_internal.constprop.0()
ldb-2.4.0/bin/default/../../tools/ldbadd.c:143: note: called from main()
#  342|   	}
#  343|   
#  344|-> 	pc = poptGetContext(argv[0], argc, argv, *popt_options,
#  345|   			    POPT_CONTEXT_KEEP_FIRST);
#  346|   

Error: VALGRIND_WARNING:
bin/ldbrename: note: while executing process 3581 with arguments: cn=Ursula Hampster,ou=Alumni Association,ou=People,o=University of Michigan,c=TEST cn=Hampster Ursula,ou=Alumni Association,ou=People,o=University of Michigan,c=TEST
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: warning[Leak_PossiblyLost]: 32 bytes in 1 blocks are possibly lost in loss record 2 of 26
/usr/lib64/libpopt.so.0.0.1: note: called from poptGetContext()
ldb-2.4.0/bin/default/../../tools/cmdline.c:344: note: called from ldb_cmdline_process_internal.constprop.0()
ldb-2.4.0/bin/default/../../tools/ldbrename.c:62: note: called from main()
#  342|   	}
#  343|   
#  344|-> 	pc = poptGetContext(argv[0], argc, argv, *popt_options,
#  345|   			    POPT_CONTEXT_KEEP_FIRST);
#  346|   

Error: COMPILER_WARNING (CWE-477):
ldb-2.4.0/common/ldb.c: scope_hint: In function ‘ldb_init’
ldb-2.4.0/common/ldb.c:120:17: warning[-Wdeprecated-declarations]: ‘tevent_loop_allow_nesting’ is deprecated
#  120 |                 tevent_loop_allow_nesting(ev_ctx);
#      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~
ldb-2.4.0/include/ldb.h:51: included_from: Included from here.
ldb-2.4.0/include/ldb_private.h:43: included_from: Included from here.
ldb-2.4.0/common/ldb.c:36: included_from: Included from here.
/usr/include/tevent.h:2304:6: note: declared here
# 2304 | void tevent_loop_allow_nesting(struct tevent_context *ev) _DEPRECATED_;
#      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
#  118|   		}
#  119|   		tevent_set_debug(ev_ctx, ldb_tevent_debug, ldb);
#  120|-> 		tevent_loop_allow_nesting(ev_ctx);
#  121|   	}
#  122|   

Error: COMPILER_WARNING (CWE-477):
ldb-2.4.0/common/ldb.c: scope_hint: In function ‘ldb_handle_new’
ldb-2.4.0/common/ldb.c:751:17: warning[-Wdeprecated-declarations]: ‘tevent_loop_allow_nesting’ is deprecated
#  751 |                 tevent_loop_allow_nesting(h->event_context);
#      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~
ldb-2.4.0/include/ldb.h:51: included_from: Included from here.
ldb-2.4.0/include/ldb_private.h:43: included_from: Included from here.
ldb-2.4.0/common/ldb.c:36: included_from: Included from here.
/usr/include/tevent.h:2304:6: note: declared here
# 2304 | void tevent_loop_allow_nesting(struct tevent_context *ev) _DEPRECATED_;
#      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
#  749|   		}
#  750|   		tevent_set_debug(h->event_context, ldb_tevent_debug, ldb);
#  751|-> 		tevent_loop_allow_nesting(h->event_context);
#  752|   	}
#  753|   

Error: COMPILER_WARNING (CWE-477):
ldb-2.4.0/tests/ldb_mod_op_test.c: scope_hint: In function ‘test_ldb_search_against_transaction’
ldb-2.4.0/tests/ldb_mod_op_test.c:1646:9: warning[-Wdeprecated-declarations]: ‘tevent_loop_allow_nesting’ is deprecated
# 1646 |         tevent_loop_allow_nesting(search_test_ctx->ldb_test_ctx->ev);
#      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
ldb-2.4.0/tests/ldb_mod_op_test.c:25: included_from: Included from here.
/usr/include/tevent.h:2304:6: note: declared here
# 2304 | void tevent_loop_allow_nesting(struct tevent_context *ev) _DEPRECATED_;
#      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
# 1644|   	struct ldb_dn *base_search_dn;
# 1645|   
# 1646|-> 	tevent_loop_allow_nesting(search_test_ctx->ldb_test_ctx->ev);
# 1647|   
# 1648|   	base_search_dn

Error: COMPILER_WARNING (CWE-477):
ldb-2.4.0/tests/ldb_mod_op_test.c: scope_hint: In function ‘test_ldb_modify_during_search’
ldb-2.4.0/tests/ldb_mod_op_test.c:1965:9: warning[-Wdeprecated-declarations]: ‘tevent_loop_allow_nesting’ is deprecated
# 1965 |         tevent_loop_allow_nesting(search_test_ctx->ldb_test_ctx->ev);
#      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
ldb-2.4.0/tests/ldb_mod_op_test.c:25: included_from: Included from here.
/usr/include/tevent.h:2304:6: note: declared here
# 2304 | void tevent_loop_allow_nesting(struct tevent_context *ev) _DEPRECATED_;
#      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
# 1963|   	}
# 1964|   
# 1965|-> 	tevent_loop_allow_nesting(search_test_ctx->ldb_test_ctx->ev);
# 1966|   
# 1967|   	ctx.basedn

Error: COMPILER_WARNING (CWE-477):
ldb-2.4.0/tests/ldb_mod_op_test.c: scope_hint: In function ‘test_ldb_modify_during_whole_search’
ldb-2.4.0/tests/ldb_mod_op_test.c:2205:9: warning[-Wdeprecated-declarations]: ‘tevent_loop_allow_nesting’ is deprecated
# 2205 |         tevent_loop_allow_nesting(search_test_ctx->ldb_test_ctx->ev);
#      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
ldb-2.4.0/tests/ldb_mod_op_test.c:25: included_from: Included from here.
/usr/include/tevent.h:2304:6: note: declared here
# 2304 | void tevent_loop_allow_nesting(struct tevent_context *ev) _DEPRECATED_;
#      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
# 2203|   	struct ldb_result *res2;
# 2204|   
# 2205|-> 	tevent_loop_allow_nesting(search_test_ctx->ldb_test_ctx->ev);
# 2206|   
# 2207|   	ctx.basedn

Error: COMPILER_WARNING (CWE-477):
ldb-2.4.0/tests/ldb_mod_op_test.c: scope_hint: In function ‘test_ldb_callback_modify_during_search’
ldb-2.4.0/tests/ldb_mod_op_test.c:2689:9: warning[-Wdeprecated-declarations]: ‘tevent_loop_allow_nesting’ is deprecated
# 2689 |         tevent_loop_allow_nesting(search_test_ctx->ldb_test_ctx->ev);
#      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
ldb-2.4.0/tests/ldb_mod_op_test.c:25: included_from: Included from here.
/usr/include/tevent.h:2304:6: note: declared here
# 2304 | void tevent_loop_allow_nesting(struct tevent_context *ev) _DEPRECATED_;
#      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
# 2687|   	}
# 2688|   
# 2689|-> 	tevent_loop_allow_nesting(search_test_ctx->ldb_test_ctx->ev);
# 2690|   
# 2691|   	ctx.basedn
